mobify/platform-scaffold
# [WEB-1220] Checkout Form Validation

Add form validation for checkout flow.

 **JIRA**: https://mobify.atlassian.net/browse/WEB-1220
 **Linked PRs**: N/A

## Changes
- Add basic form validation for checkout flow & tax estimation
- Remove runtype errors in checkout flow
- Remove login button on empty cart if already logged in
- Add CC validation with luhn algorithm & CC expiry validation
- `Calculate` button should appear on cart summary to calculate tax when there is no zip code present.

## Todos

## How to test-drive this PR
- Make sure your node_modules is pretty fresh (works with latest develop)
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Run through the checkout flow and notice:
    1. There are no longer runtype validation errors every page
    1. Checkout will not allow you to proceed with empty inputs.
    1. Checkout payment will not let you proceed with a credit card number that does not pass Luhn's checksum algorithm. (a card # that does work: 40303000000... any amount of 0's after will pass)


Existing Issues that were noted on the ticket that are not fixed due to scope:
- Saved credit cart not appearing (logged in) 

Although I don't see anywhere we can save a credit card?.. so maybe this isn't an issue until we have saved credit accounted for.
